### DISCUSSION TITLE
regex issues

### DISCUSSION THREAD

**Discussion by ghost** [4sval/FModel#524] [Q&A]
i dont know how to explain it, but package searching works fine, but trying to use common regex sometimes returns no results, and a blank prompt shows no results. it is hard to understand what "regex" is actually in used because dumping folder/file structure and using other tools returns an absurd amount of results that fmodel appears to fail at doing

> `dialog/idle_bandit` (regex off)
> TestGame/Content/Data/Dialog/Idle_BanditsNear.uasset
> TestGame/Content/Data/Dialog/Idle_BanditsKilled.uasset

> `dialog.*bandit` (regex on)
> `no folder found in archives, make sure you loaded at least one`
> (turn back off, returns paths)

> `dialog.*b` (regex on)
> incorrect paths, captures "dialog" but "any all" (.*) fails

> `dialog.*ba` (regex on)
> still incorrect paths

> `dialog.*ban` (regex on)
> `no folder found in archives, make sure you loaded at least one`

on unix using various cli utilities and graphical text editors, their "regex" appears to work as intended, even testing windows10 "findstr" regex works better than fmodel

this has been an issue long before version 4.0.0 and i thought it was just me not knowing my regex, but this feature actually impedes a smooth workflow.
this is not really explained or shown in discussion, issues, or pull requests so i had always assumed it was "working as intended"

current fmodel (c6c8cf2) 4.4.4.0

are we doing something wrong, or is there specific patterns for fmodel

**Comment by LongerWarrior** [comment]
seems like a bug with MatchCase state, for now click on it to get correct results
https://github.com/4sval/FModel/blob/d132bc46e279f1bcb8e71737827c05e26f3f8707/FModel/ViewModels/SearchViewModel.cs#L61

**Comment by ghost** [comment]
that is interesting...
